chain-abstraction/sol-implementation/edge_challenge_manager.go

```diff
@@ -29,6 +29,19 @@ func (e *specEdge) GetType() protocol.EdgeType {
 	return protocol.EdgeType(e.inner.EType)
 }
 
+func (e *specEdge) GetChallengeLevel() (protocol.ChallengeLevel, error) {
+	switch protocol.EdgeType(e.inner.EType) {
+	case protocol.BlockChallengeEdge:
```

Comment:
The reason we go from lowest challenge level being 0 instead of 2 is to make our code a lot more readable. If we flipped the order, we would need to do a lot of backwards for loops instead of simple range loops over slices

Reply:
This is nice context to add in a godoc somewhere